src/api/index.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import { HttpMethod, callApi, logOutFromServer } from '.';
+import {
+  initialState as defaultApiState,
+  actions as apiActions,
+} from '../reducers/api';
+import configureStore from '../configureStore';
+
+describe(__filename, () => {
+  const getApiState = ({ authToken = '12345' } = {}) => {
+    const store = configureStore();
+
+    store.dispatch(apiActions.setAuthToken({ authToken }));
+    const { api } = store.getState();
+
+    return api;
+  };
+
+  describe('callApi', () => {
+    const callApiWithDefaultApiState = (params = {}) => {
+      return callApi({ apiState: defaultApiState, endpoint: '/', ...params });
+    };
+
+    it('calls the API', async () => {
+      await callApiWithDefaultApiState({ endpoint: '/foo/' });
+
+      expect(fetch).toHaveBeenCalledWith(`/api/v4/foo/`, {
+        headers: {},
+        method: 'GET',
+      });
+    });
+
+    it('adds a trailing slash to the endpoint if there is none', async () => {
+      await callApiWithDefaultApiState({ endpoint: '/foo' });
+
+      expect(fetch).toHaveBeenCalledWith(`/api/v4/foo/`, expect.any(Object));
+    });
+
+    it('adds a leading slash to the endpoint if there is none', async () => {
+      await callApiWithDefaultApiState({ endpoint: 'foo/' });
+
+      expect(fetch).toHaveBeenCalledWith(`/api/v4/foo/`, expect.any(Object));
+    });
+
+    it('accepts an HTTP method', async () => {
+      const method = HttpMethod.POST;
+
+      await callApiWithDefaultApiState({ endpoint: '/foo/', method });
+
+      expect(fetch).toHaveBeenCalledWith(
```

Comment:
A helper could be used here as well, `toHaveBeenCalledWithProperties`, which ignores the URL (by passing `expect.any(String)` as the first argument), and accepts an object with the expected properties. This could be combined with my suggestion about a default properties object which could be used to compose the argument to this new helper.